cmd/kubectl-gadget/main.go

```diff
@@ -125,7 +128,22 @@ func main() {
 	}
 
 	if !skipInfo {
+		kubectlGadgetVersion := version.Version()
 		grpcRuntime.InitDeployInfo()
+		info, err := deployinfo.Load()
+		if err != nil {
+			log.Warnf("failed to load deploy info: %s", err)
+		}
+		serverSemver, err := semver.Make(info.ServerVersion)
+		if err != nil {
+			log.Errorf("Failed to create a semantic version object for the gadget pods version: %s", err)
+		}
+		if !kubectlGadgetVersion.Equals(serverSemver) {
+			log.Warnf(
+				"version skew detected (gadget pods v%s vs kubectl-gadget v%s), use 'kubectl gadget deploy' to fix it.\n",
+				serverSemver, kubectlGadgetVersion,
+			)
```

Comment:
I think all of this can be factorized out on a new helper function that is called from kubetl-gadget and gadgetctl.

Reply:
Should I put the helper function in ./internal/version dir in a seperate file? I was thinking a name like "versionChecker" maybe.

Reply:
I think it can go to `cmd/common/utils`.